cpp/src/parquet/column_reader.cc

```diff
@@ -1005,7 +1005,7 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatchWithDictionary(
 
   // Read dictionary indices.
   *indices_read = ReadDictionaryIndices(indices_to_read, indices);
-  int64_t total_indices = std::max(num_def_levels, *indices_read);
+  int64_t total_indices = std::max<int>(num_def_levels, *indices_read);
```

Comment:
should this be int64_t for the template parameter?

Reply:
done